### PR TITLE
Fix exception handling in `load_hf_model_from_task`

### PR DESCRIPTION
**Pull request by jambayk**
## Describe your changes
`load_hf_model_from_task` catches `OSError,ValueError` to allow iterating over potentially multiple candidate classes. However, for most cases, there is only one model class to try and there are valid errors like device OOM, unsupported kwargs, that need to be rethrown. 
Currently, the model loading silently fails and the caller gets `None` as the model. This is unexpected behavior and makes it hard to debug without error logs or failure with traceback.
 
## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/model/utils/hf_utils.py [hunk 1]
```diff
@@ -41,22 +37,32 @@ def load_hf_model_from_task(task: str, name: str, **kwargs):
...
-    model = None
+    exceptions = []
     for model_class in class_tuple:
         try:
             model = model_class.from_pretrained(name, **kwargs)
             logger.debug("Loaded model %s with name_or_path %s", model_class, name)
             return model
-        except (OSError, ValueError):
+        except (OSError, ValueError) as e:
+            if len(class_tuple) == 1:
```
Comment (guotuofeng) [inline olive/model/utils/hf_utils.py:47]: what about we raise the exception if it is the last item in the tuple?
```python
for i, model_class in enumerate(class_tuple):
      except:
        if i == len(class_tuple) - 1:
           raise
```

File: olive/model/utils/hf_utils.py [hunk 2]
```diff
@@ -37,26 +33,42 @@ def load_hf_model_from_task(task: str, name: str, **kwargs):
...
+                # error could be device OOM, device_map: "auto" not supported, etc
+
+                # len(class_tuple) > 1: not common - image-segmentation, conversational, etc
+                # there is no easy way to get tracebacks for earlier failures, so just raise from last
+                raise
             # the ValueError need to be caught since there will be multiple model_class for single task.
             # if the model_class is not the one for the task, it will raise ValueError and
             # next model_class will be tried.
-            continue
+            logger.debug(
```
Comment (guotuofeng) [inline olive/model/utils/hf_utils.py:54]: warning?

File: olive/model/utils/hf_utils.py [hunk 3]
```diff
@@ -37,26 +33,42 @@ def load_hf_model_from_task(task: str, name: str, **kwargs):
...
+                "Failed to load model %s with name_or_path %s.\n kwargs: %s.\n Exception raised: %s",
+                model_class,
+                name,
+                kwargs,
+                e,
+            )
+
+    if model is None:
```
Comment (guotuofeng) [inline olive/model/utils/hf_utils.py:62]: we can safely remove the model None check and raise of ValueError?

Comment (jambayk) [reply olive/model/utils/hf_utils.py]: updated

**Review by guotuofeng — APPROVED** [review]

File: olive/model/utils/hf_utils.py (hunk 2 above)
Comment (jambayk) [reply olive/model/utils/hf_utils.py]: changed to info

File: olive/model/utils/hf_utils.py (hunk 1 above)
Comment (jambayk) [reply olive/model/utils/hf_utils.py]: done

**Pull request merged**